src/blockdata/transaction.rs

```diff
@@ -381,6 +382,31 @@ impl Sequence {
     pub fn to_consensus_u32(self) -> u32 {
         self.0
     }
+
+    /// Creates a [`relative::LockTime`] from this [`Sequence`] number.
+    #[inline]
+    pub fn to_relative_lock_time(&self) -> Option<relative::LockTime> {
+        use crate::locktime::relative::{LockTime, Height, Time};
+
+        if !self.is_relative_lock_time() {
+            return None;
+        }
+
+        let lock_value = self.low_u16();
+
+        if self.is_time_locked() {
+            Some(LockTime::from(Time::from_512_second_intervals(lock_value)))
+        } else {
+            Some(LockTime::from(Height::from(lock_value)))
+        }
+    }
+
+    /// Returns the low 16 bits from sequence number.
+    ///
+    /// BIP-68 only uses the low 16 bits for relative lock value.
+    fn low_u16(&self) -> u16 {
+        (self.0 & 0x0000_FFFF) as u16
```

Comment:
In fb5abb43711a9f39e62f0e7b25dda8f85b32e98b:

The masking seems unnecessary.

Reply:
I did it on purpose to make the `as u16` explicitly safe because of the talk recently (@dpc and @Kixunil) about not use `as`. Can remove it (or leave it and document why).

Reply:
I have a mild preference for removing it. I think the doccomment already justifies the loss of bits. The masking adds to the cognitive load.

Reply:
Sure thing, thanks.